lib/ops/opsservice/update.go

```diff
@@ -70,13 +69,59 @@ func (o *Operator) RotateSecrets(req ops.RotateSecretsRequest) (*ops.RotatePacka
 	return resp, nil
 }
 
-// RotatePlanetConfig rotates planet configuration package for the server specified in the request
-func (o *Operator) RotatePlanetConfig(req ops.RotatePlanetConfigRequest) (*ops.RotatePackageResponse, error) {
-	operation, err := o.GetSiteOperation(req.SiteOperationKey())
+// RotateTeleportConfig generates teleport configuration for the server specified in the provided request
+func (o *Operator) RotateTeleportConfig(req ops.RotateTeleportConfigRequest) (nodeConfig *ops.RotatePackageResponse, err error) {
+	if err := req.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	operation, err := o.GetSiteOperation(req.Key)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	nodeProfile, err := o.getNodeProfile(*operation, req.Server)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	node := &ProvisionedServer{
+		Server:  req.Server,
+		Profile: *nodeProfile,
+	}
+
+	cluster, err := o.openSite(req.Key.SiteKey())
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	nodeConfigPackage := req.Node
+	if nodeConfigPackage == nil {
+		nodeConfigPackage, err = cluster.teleportNodeConfigPackage(node)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	ctx, err := cluster.newOperationContext(*operation)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	masterIPs := (storage.Servers)(req.Servers).MasterIPs()
+	if len(masterIPs) == 0 {
+		return nil, trace.NotFound("no master server found")
+	}
+
+	nodeConfig, err = cluster.getTeleportNodeConfig(ctx, masterIPs[0], *nodeConfigPackage, node)
```

Comment:
@r0mant not sure if I needed to pull the change to use all master addresses - do you think it's necessary for 5.2.x?

Reply:
I think it depends on what will happen with the nodes if this `masterIPs[0]` master goes down / is removed. Presumably they won't be able to contact auth server anymore? We can address this in another PR though as it seems tangential.